### PR TITLE
add data set for KNN

### PR DESCRIPTION
**Pull request by MikeLing**
This pr is for shogun-toolbox/shogun#3513 and support low-dimension data comparison. This Letter Image Recognitiondata set is to identify each of a large number of black-and-white  rectangular pixel displays as one of the 26 capital letters in the English  alphabet.  The character images were based on 20 different fonts and each letter within these 20 fonts was randomly distorted to produce a file of 20,000 unique stimuli.  Each stimulus was converted into 16 primitive numerical attributes (statistical moments and edge counts) which were then scaled to fit into a range of integer values from 0 through 15.  We typically train on the first 16000 items and then use the resulting model to predict the letter category for the remaining 4000.  See the article cited above for more details.

For more information about data set, see http://archive.ics.uci.edu/ml/machine-learning-databases/letter-recognition/letter-recognition.names. @karlnapf please review this pr. :)

### CODE REVIEW

**Comment by karlnapf** [comment]
* How big is the file? Seems huge ... Can we do smaller to reduce overhead?
 * The details you gave in the PR description need to be in some comment in the file, or a description file, so that people can read them without having to look at the PR
 * Needs approval from another dev

**Comment by MikeLing** [comment]
> How big is the file? Seems huge ... Can we do smaller to reduce overhead?
the file size is 713KB, and I just try to use only half of that data set to test KNN. Here is what I got:

Standard KNN took 1.3338s
Covertree KNN took 0.3085s
KDTree KNN took 0.5545s

> The details you gave in the PR description need to be in some comment in the file, or a description file

Hmmm, is that mean I need to create another file as description for data set? Like add file named **letter-recognition-description.txt**  in this pr? I guess this file is enough to go and I can just copy this whole file in this pr or just ass a reference in the notebook.

> Needs approval from another dev

sure thing, I will ask review again after address those issues above :) 
Thank you

**Comment by karlnapf** [comment]
Size is OK then, but we *do* need a description for it

**Comment by MikeLing** [comment]
@karlnapf , I add a file as description of letter-recognition data set and only use half of the original data set :)

File: multiclass/description-of-letter-recognition.txt [hunk 1]
```diff
@@ -0,0 +1,62 @@
+1. Title: Letter Image Recognition Data
```
Comment (karlnapf) [inline multiclass/description-of-letter-recognition.txt]: can you make "description" a suffix rather than a prefix, so the files are sorted together?

**Comment by karlnapf** [comment]
Fine to merge from my side...
@vigsterkr ?

**Pull request merged**